decode.c

```diff
@@ -2,12 +2,20 @@
 #include <stdbool.h>
 #include <stddef.h>
 #include <stdint.h>
+#include <string.h>
 
 #include <fadec.h>
 
 
+#ifdef _MSC_VER
```

Comment:
I'd prefer these checks to be the other way: check for `__GNUC__`, and if not, use the more generic version.